tests/js/plugin/texthighlight-test.coffee

```diff
@@ -0,0 +1,78 @@
+assert = chai.assert
+sinon.assert.expose(assert, prefix: '')
+
+# In order to be able to create highlights,
+# the Annotator.TextHighlight class must exist.
+# This class is registered then the TextHighlights plugin
+# is initialized, so we will do that.
+th = new Annotator.Plugin.TextHighlights()
+th.pluginInit()
+
+describe 'Annotator.Plugin.TextHighlight', ->
+  sandbox = null
+  jqElement = null
+  scrollTarget = null
+
+  createTestHighlight = ->
+    anchor =
+      id: "test anchor"
+      annotation: "test annotation"
+      anchoring:
+        id: "test anchoring manager"
+        annotator:
+          id: "test annotator"
+          element:
+            delegate: sinon.spy()
+
+    new Annotator.TextHighlight anchor, "test page", "test range"
+
+  beforeEach ->
+    sandbox = sinon.sandbox.create()
+    sandbox.stub Annotator.TextHighlight, 'highlightRange',
+      (normedRange, cssClass) -> "test highlight span"
+
+    sandbox.stub Annotator.$.fn, "init", (selector, context) ->
+      jqElement =
+        selector: selector
+        data: sinon.spy()
+        scrollintoview: sinon.spy (options) ->
+          scrollTarget = this.selector
+          options?.complete?()
+
+  afterEach ->
+    sandbox.restore()
+    scrollTarget = null
+
+  describe "constructor", ->
+    it 'wraps a highlight span around the given range', ->
+      hl = createTestHighlight()
+      assert.calledWith Annotator.TextHighlight.highlightRange, "test range"
+
+    it 'stores the created highlight spans in _highlights', ->
```

Comment:
> You're clearly not talking about CSS visibility, since words merely hidden that way are clearly "in the document", so I think you're talking about PDF.js, perhaps, or more generally, document formats projected into HTML by some viewer application. 

Exactly. As I [said](https://github.com/hypothesis/h/pull/1950/files#r24718466) right at the top: 

> This distinction is important on all reader/viewer platforms where only a part of the document is rendered to the DOM at any single time.

> In which case, the argument seems circular: things are not `Element`s because they are not `Element`s.

That doesn't really represent my argument. My argument was that it makes little sense for us to try to represent the anchors as DOM Elements as long as the rest of the document is not present in the DOM either.

> Yet we use them an awful lot like `Element`s. We add and remove them from the rendering tree; they are sometimes attached and sometimes not; we sometimes wish to focus on them, scroll them into view, find text within them, etc.

What you are describing here is definitely not specific to anchors or highlights; instead, these traits are generally true for all the contents of the document; yet, despite all of these (valid) reasons, they are *not* represented as Elements ... at least in PDF.js

> Could not PDF.js have avoided implementing their own search were there a way to present detached elements to the browser's find feature? I don't know. 

I think they didn't want to restrict the allowed document size to the size that the browser can (reasonably) handle. But this is not only about PDF.js; as far as I know, neither Google Docs, nor Scribd, nor any other similar services add elements for the content that is outside of the section currently rendered. Instead of this, they all provide custom search features that search in the (hidden) corpus.

I think this shows that the currently available browser APIs don't really cover this use-case.

(However, it would be great to help designing an API that would make it possible to do this in a standard-driven, unified way.)

> But that's all got to do with hidden content anyway, and we don't even need to discuss hidden hidden content to make the case that, at the least, visible highlights could be `Element`s and are more than likely associated with one or more `Element`s.

Yes, visible highlights are associated with  `Elements`. Yes, maybe we could represent them as a new type of `Elements` ... but I don't know too much about those.